core/crypto/rsa_common.go

```diff
@@ -11,10 +11,12 @@ import (
 const WeakRsaKeyEnv = "LIBP2P_ALLOW_WEAK_RSA_KEYS"
 
 var MinRsaKeyBits = 2048
+var MaxRsaKeyBits = 8192
```

Comment:
Does this need to be configurable?

```suggestion
const maxRsaKeyBits = 8192
```